staging/src/k8s.io/kube-aggregator/pkg/controllers/autoregister/autoregister_controller.go

```diff
@@ -54,7 +54,7 @@ const (
 // adding and removing APIServices
 type AutoAPIServiceRegistration interface {
 	// AddAPIServiceToSyncOnStart adds an API service to sync on start.
-	AddAPIServiceToSyncOnStart(in *v1.APIService)
+	AddAPIServiceToSyncOnStart(in *v1.APIService, hash string)
```

Comment:
When I made the suggestion of moving the hash, I was thinking that rather than having it supplied by the called of `AddAPIServiceToSync`, the hash would be fetched from E-Tag within `checkAPIService`.

If done that way I am actually wondering if it would work also for CRDs since in the current implementation the CRDRegistrationController also calls autoregister_controller's `AddAPIServiceToSync`.

Reply:
We would need to supply the handlers to the autoregister_controller so that a request may be made to fetch the hash. I don't have a strong preference, happy to make the change if that's more desirable.

/cc @apelisse @kevindelgado

Reply:
we can merge and decide later?